lib/cmd/compile/fonts.js

```diff
@@ -0,0 +1,192 @@
+  fontFormats = ['woff', 'woff2', 'otf', 'ttf'],
+  variables = {
+    // asset host and path are set in different environments when using separate servers/subdomains for assets
+    'asset-host': process.env.CLAYCLI_COMPILE_ASSET_HOST ? process.env.CLAYCLI_COMPILE_ASSET_HOST.replace(/\/$/, '') : '',
+    'asset-path': process.env.CLAYCLI_COMPILE_ASSET_PATH,
+    // these arguments allow setting default env variables
+    minify:  process.env.CLAYCLI_COMPILE_MINIFIED || process.env.CLAYCLI_COMPILE_MINIFIED_FONTS,
+    inlined: process.env.CLAYCLI_COMPILE_INLINED_FONTS,
+    linked: process.env.CLAYCLI_COMPILE_LINKED_FONTS
+  };
+
+/**
+ * default 'linked' to true, if inlined is NOT set
+ * @param  {boolean|undefined} linked
+ * @param  {boolean} inlined
+ * @return {boolean}
+ */
+function getLinkedSetting(linked, inlined) {
+  // default linked to true UNLESS inlined is set (and linked isn't)
+  if (typeof linked === 'undefined' && inlined) {
+    // inlined is set, so don't link fonts
+    return false;
+  } else if (typeof linked === 'undefined') {
+    // inlined isn't set, so link fonts by default
+    return true;
+  } else {
+    return linked;
+  }
+}
+
+/**
+ * get name, style, and weight based on a font's filename
+ * @param  {array} fontArray e.g. ['georgiapro', 'bold', 'italic']
+ * @return {object} w/ { name, style, weight } css declarations
+ */
+function getFontAttributes(fontArray) {
+  let name = fontArray[0], // e.g. georgiapro, note: font families are case insensitive in css
+    weight, style;
+
+  if (fontArray.length === 3) {
+    // name-weight-style
+    weight = _.findKey(fontWeights, (val) => _.includes(val, fontArray[1]));
+    style = _.find(fontStyles, (val) => val === fontArray[2]);
+  } else if (fontArray.length === 2 && _.find(fontStyles, (val) => val === fontArray[1])) {
+    // name-style (note: checking for style is faster than weight, so we do that first)
+    style = _.find(fontStyles, (val) => val === fontArray[1]);
+  } else if (fontArray.length === 2) {
+    // name-weight
+    weight = _.findKey(fontWeights, (val) => _.includes(val, fontArray[1]));
+  } // else it's just the name
+
+  return {
+    name: `font-family: "${name}"; `, // note: trailing spaces so they can all be concatenated
+    weight: weight ? `font-weight: ${weight}; ` : '',
+    style: style ? `font-style: ${style}; ` : ''
+  };
+}
+
+/**
+ * get filename, file format, font name, font style, and font weight
+ * note: the returned 'css' is the beginning of the @font-face declaration
+ * for both inlinedd and linked fonts
+ * @param  {object} file
+ * @param  {string} styleguide
+ * @param  {boolean} isInlined
+ * @return {string} @font-face declaration
+ */
+function getFontCSS(file, styleguide, isInlined) {
+  const ext = path.extname(file.path), // e.g. '.woff'
+    fileName = path.basename(file.path), // e.g. 'GeorgiaProBold.woff'
+    fontAttrs = getFontAttributes(path.basename(file.path, ext).toLowerCase().split('-')),
```

Comment:
If you don't use `fontAttrs` again I feel like it would be better to change `getFontAttributes` to `getFontFace` and include the `@font-face { ${fontAttrs.name}${fontAttrs.style}${fontAttrs.weight}` there

Reply:
eh, `fontAttrs` is something that looks like `['helvetica', 'bold', 'italic']`. Name is one of the attributes, as is other stuff.